abnf/odata-aggregation-abnf.txt

```diff
@@ -110,25 +110,29 @@ hierarchyTrafo  = ancestorsTrafo
 preservingTrafos = preservingTrafo *( "/" preservingTrafo )
 hierarchyTrafos  = hierarchyTrafo *( "/" hierarchyTrafo )
 
-aggregateTrafo  = %s"aggregate" OPEN BWS aggregateExpr *( BWS COMMA BWS aggregateExpr ) BWS CLOSE
-aggregateExpr   = aggregatableExpr aggregateWith [ aggregateFrom ] asAlias
-                / ( aggrPathPrefix / aggrCastPath ) nonprimAggWith [ aggregateFrom ] asAlias
-                / aggregateCount [ aggregateFrom ] asAlias
-                / aggregateCustom [ [ customFrom ] asAlias ]
-aggregateWith   = RWS %s"with" RWS aggregateMethod
-nonprimAggWith  = RWS %s"with" RWS nonprimAggMethod
-aggregateFrom   = RWS %s"from" RWS groupingProperties aggregateWith [ aggregateFrom ]
-customFrom      = RWS %s"from" RWS groupingProperties [ aggregateWith ] [ customFrom ]
-aggregateMethod = %s"sum"
-                / %s"min"
-                / %s"max"
-                / %s"average"
-                / nonprimAggMethod
+aggregateTrafo   = %s"aggregate" OPEN BWS aggregateExpr *( BWS COMMA BWS aggregateExpr ) BWS CLOSE
+aggregateExpr    = ( aggrPathPrefix / aggrCastPath ) nonprimAggWith [ aggregateFrom ] asAlias
+                 / aggregatableExpW [ aggregateFrom ] asAlias
+                 / aggregateCount [ aggregateFrom ] asAlias
+                 / aggregateCustom [ [ customFrom ] asAlias ]
+aggregatableExpr = commonExpr ; resulting in an aggregatable value
+aggregatableExpW = aggregatableExpr aggregateWith
+                 / [ aggrCastPath "/" ] aggrPrimPath aggregateWith

```

Comment:
Neither
```
aggregatableExpW = ( aggregatableExpr / [ aggrCastPath "/" ] aggrPrimPath )
                   aggregateWith
```
nor
```
aggregatableExpW = ( [ aggrCastPath "/" ] aggrPrimPath / aggregatableExpr )
                   aggregateWith
```
works, because either way some of the test cases fail.

What does this imply for the ability of a parser to distinguish between aggregate expressions of type 1 (`[ aggrCastPath "/" ] aggrPrimPath`) and type 2 (`aggregatableExpr`)?

Reply:
It can't because both rules have `simpleIdentifier` as a common subset.